editor/src/messages/portfolio/document/utility_types/network_interface.rs

```diff
@@ -6423,17 +6739,59 @@ pub struct DocumentNodePersistentMetadataInputNames {
 
 impl From<DocumentNodePersistentMetadataInputNames> for DocumentNodePersistentMetadata {
 	fn from(old: DocumentNodePersistentMetadataInputNames) -> Self {
-		let input_properties = old
-			.reference
-			.as_ref()
-			.and_then(|reference| resolve_document_node_type(reference))
-			.map(|definition| definition.node_template.persistent_node_metadata.input_properties.clone())
-			.unwrap_or(old.input_names.into_iter().map(|name| (name.as_str(), "").into()).collect());
+		DocumentNodePersistentMetadata {
+			input_metadata: Vec::new(),
+			..old.into()
+		}
+	}
+}
 
+#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct DocumentNodePersistentMetadataPropertiesRow {
+	pub reference: Option<String>,
+	#[serde(default)]
+	pub display_name: String,
+	pub input_properties: Vec<PropertiesRow>,
+	#[serde(deserialize_with = "migrate_output_names")]
+	pub output_names: Vec<String>,
+	#[serde(default = "return_true")]
+	pub has_primary_output: bool,
+	#[serde(default)]
+	pub locked: bool,
+	#[serde(default)]
+	pub pinned: bool,
+	pub node_type_metadata: NodeTypePersistentMetadata,
+	pub network_metadata: Option<NodeNetworkMetadata>,
+}
+
+#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct PropertiesRow {
+	pub input_data: HashMap<String, Value>,
+	pub widget_override: Option<String>,
+	#[serde(skip)]
+	pub input_name: String,
+	#[serde(skip)]
+	pub input_description: String,
+}
+
+impl From<DocumentNodePersistentMetadataPropertiesRow> for DocumentNodePersistentMetadata {
+	fn from(old: DocumentNodePersistentMetadataPropertiesRow) -> Self {
+		let mut input_metadata = Vec::new();
+		for properties_row in old.input_properties {
+			input_metadata.push(InputMetadata {
+				persistent_metadata: InputPersistentMetadata {
+					input_data: properties_row.input_data,
+					widget_override: properties_row.widget_override,
+					input_name: properties_row.input_name,
+					input_description: properties_row.input_description,
+				},
+				..Default::default()
+			})
+		}
 		DocumentNodePersistentMetadata {
 			reference: old.reference,
 			display_name: old.display_name,
-			input_properties,
+			input_metadata: Vec::new(),
```

Comment:
when upgrading from an old file, replace the input metadata with an empty vec, which will be populated based on the definition in validate_input_metadata